lib/hanami/routing/route.rb

```diff
@@ -50,7 +50,8 @@ def generate(resolver, options = {}, &endpoint)
       # @since 0.2.0
       # @api private
       def nested_router
-        dest.routes if dest.respond_to?(:routes)
+        dest.routes if dest.respond_to?(:routes) &&
+          dest.routes.is_a?(Hanami::Router)
```

Comment:
> it may potentially make the hanami routes output look strange or screwed up.

We don't expect to support what we don't know, but to make it open for others to provide a routes good inspection. If you mount a non "inspectable" Rack app, it's not a big deal if the routes printing isn't pretty.